server/src/services/series.js

```diff
@@ -37,6 +39,86 @@ const aggregateSeries = async (seriesIds) => {
     ]);
 };
 
+const filterSeries = async ({ name, status, genre }, pageNumber, pageLimit) => {
+    const aggregationQuery = [];
+
+    name && aggregationQuery.push({ $addFields: { searchIndex: { $indexOfCP: ["$name", String(name)] } } }, { $match: { searchIndex: { $ne: -1 } } });
+    status && aggregationQuery.push({ $match: { status: { $eq: status } } });
+    aggregationQuery.push(...lookupGenres(), {
+        $replaceWith: {
+            $setField: {
+                field: "genres",
+                input: "$$ROOT",
+                value: {
+                    $reduce: {
+                        input: "$genres",
+                        initialValue: [],
+                        in: { $concatArrays: ["$$value", ["$$this.name"]] }
+                    }
+                }
+            }
+        }
+    });
+    genre && aggregationQuery.push({ $match: { genres: { $all: genre } } });
+
+    const data = await Series.aggregate([...aggregationQuery, { $skip: pageLimit * (parseInt(pageNumber) - 1) }, { $limit: pageLimit }]);
+    const totalAmount = await Series.aggregate([...aggregationQuery, { $count: "count" }]);
+
+    return { data, totalAmount };
+};
+
+const getMostWatchedSeries = async (pageNumber, pageLimit) => {
+    const result = await WatchLists.aggregate([
+        ...lookupSeriesFromEpisode(),
+        ...lookupGenres(),
+        { $skip: pageLimit * (parseInt(pageNumber) - 1) },
+        { $limit: pageLimit }
+    ]);
+
+    return result;
+};
+
+const getCommonSeriesAmongFollowing = async (email, userWatchList, pageNumber, pageLimit) => {
+    const following = await searchFollowers(email);
+    
+    const followingWatchList = await WatchLists.aggregate([
+        { $match: { email: { $in: following.map(follower => follower.email_to) } } },
+        ...lookupSeriesFromEpisode(),
+        { $match: {name: { $nin: userWatchList }}},
+        ...lookupGenres(),
```

Comment:
I don't see group by series and sort based on their appearance in others' watchlists (in Spring Boot it had a DESC sort based on `common_series.count`
Didn't manually check it since it probably doesn't work as expected, I'll go back to that when you update the implementation